airflow/api/common/trigger_dag.py

```diff
@@ -39,6 +39,7 @@ def _trigger_dag(
     conf: dict | str | None = None,
     execution_date: datetime | None = None,
     replace_microseconds: bool = True,
+    triggered_by: DagRunTriggeredByType | None = None,
```

Comment:
```suggestion
    triggered_by: DagRunTriggeredByType,
```

Same as mentioned previously, I think we should make this strictly required, and keep the None case only for legacy records. There are other arguments in the call chain that should be changed as well.